File: app/src/main/java/com/github/pockethub/android/ui/item/news/IssuesEventItem.java
package com.github.pockethub.android.ui.item.news;

import android.support.annotation.NonNull;
import android.text.TextUtils;
import android.view.View;

import com.github.pockethub.android.ui.StyledText;
import com.github.pockethub.android.ui.view.OcticonTextView;
import com.github.pockethub.android.util.AvatarLoader;
import com.meisolsson.githubsdk.model.GitHubEvent;
import com.meisolsson.githubsdk.model.Issue;
import com.meisolsson.githubsdk.model.payload.IssuesPayload;

public class IssuesEventItem extends NewsItem {

    public IssuesEventItem(AvatarLoader avatarLoader, GitHubEvent dataItem) {
        super(avatarLoader, dataItem);
    }

    @Override
    public void bind(@NonNull NewsItem.ViewHolder viewHolder, int position) {
        super.bind(viewHolder, position);
        IssuesPayload payload = (IssuesPayload) getData().payload();
        IssuesPayload.Action action = payload.action();

        if (action != null) {
            switch (action) {
                case Opened:
                    viewHolder.icon.setText(OcticonTextView.ICON_ISSUE_OPEN);
                    break;
                case Reopened:
                    viewHolder.icon.setText(OcticonTextView.ICON_ISSUE_REOPEN);
                    break;
                case Closed:
                    viewHolder.icon.setText(OcticonTextView.ICON_ISSUE_CLOSE);
                    break;
                default:
                    viewHolder.icon.setVisibility(View.GONE);
            }
        }

        StyledText main = new StyledText();
        boldActor(main, getData());

        Issue issue = payload.issue();
        main.append(' ');
        main.append(action.name().toLowerCase());
        main.append(' ');
        main.bold("issue " + issue.number());
        main.append(" on ");

        boldRepo(main, getData());

        StyledText details = new StyledText();
        appendText(details, issue.title());

        if (TextUtils.isEmpty(details)) {
            viewHolder.details.setVisibility(View.GONE);
        } else {
            viewHolder.details.setText(details);
        }

        viewHolder.event.setText(main);
    }
}

File: app/src/main/java/com/github/pockethub/android/ui/item/news/IssuesEventItem.kt
package com.github.pockethub.android.ui.item.news

import android.text.TextUtils
import android.view.View
import androidx.core.text.bold
import androidx.core.text.buildSpannedString
import com.github.pockethub.android.ui.issue.IssuesViewActivity
import com.github.pockethub.android.ui.view.OcticonTextView
import com.github.pockethub.android.util.AvatarLoader
import com.github.pockethub.android.util.ConvertUtils
import com.github.pockethub.android.util.android.text.clickable
import com.meisolsson.githubsdk.model.GitHubEvent
import com.meisolsson.githubsdk.model.payload.IssuesPayload
import com.xwray.groupie.kotlinandroidextensions.ViewHolder
import kotlinx.android.synthetic.main.news_item.*

class IssuesEventItem(
        avatarLoader: AvatarLoader,
        gitHubEvent: GitHubEvent
) : NewsItem(avatarLoader, gitHubEvent) {

    override fun bind(holder: ViewHolder, position: Int) {
        super.bind(holder, position)
        val payload = gitHubEvent.payload() as IssuesPayload?
        val action = payload?.action()

        if (action != null) {
            holder.tv_event_icon.visibility = View.VISIBLE
            when (action) {
                IssuesPayload.Action.Opened ->
                    holder.tv_event_icon.text = OcticonTextView.ICON_ISSUE_OPEN
                IssuesPayload.Action.Reopened ->
                    holder.tv_event_icon.text = OcticonTextView.ICON_ISSUE_REOPEN
                IssuesPayload.Action.Closed ->
                    holder.tv_event_icon.text = OcticonTextView.ICON_ISSUE_CLOSE
                else -> holder.tv_event_icon.visibility = View.GONE
            }
        }

        val issue = payload?.issue()

        val details = buildSpannedString {
            appendText(this, issue?.title())
        }

        if (TextUtils.isEmpty(details)) {
            holder.tv_event_details.visibility = View.GONE
        } else {
            holder.tv_event_details.visibility = View.VISIBLE
            holder.tv_event_details.text = details
        }

        holder.tv_event.text = buildSpannedString {
            val context = holder.root.context
            boldActor(context, this, gitHubEvent)
            append(" ${action?.name?.toLowerCase()} ")
            bold {
                clickable(onClick = {
                    val repository = ConvertUtils.eventRepoToRepo(gitHubEvent.repo())
                    context.startActivity(IssuesViewActivity.createIntent(issue!!, repository))
                }) {
                    append("issue " + issue?.number())
                }
            }
            append(" on ")
            boldRepo(context, this, gitHubEvent)
        }
    }
}
